### ISSUE TITLE
`__builtin_offsetof` unsupported

### ISSUE THREAD

**Issue by andrewvaughanj** [ltcmelo/psychec#31]
#### GCC version:
```bash
gcc (SUSE Linux) 10.2.1 20201202 [revision e563687cf9d3d1278f45aaebd03e0f66531076c9]
```

#### clang version:
```bash
clang version 11.0.1
```

#### Output of `gcc -Werror -Wall -Wextra -Wpedantic -pedantic -std=c11 -c $FNAME`
```bash
(return code 0)
```

#### Output of `clang -Werror -Wall -Wextra -Wpedantic -pedantic -std=c11 -c $FNAME`
```bash
(return code 0)
```

#### Output of `cnip $FNAME`
```bash
memset.c:4:28 error: expected expression got `struct'
  return __builtin_offsetof(struct s, i);
                            ^

(return code 0)
```

#### Content of file
```bash
struct s {
  char i;
};
int offset() {
  return __builtin_offsetof(struct s, i);
}
```

#### `psychec` version
```bash
commit d10dc1ddfb7ea845f59274a4d6ca5d021ab31423
Author: Leandro T. C. Melo <LTCMELO@GMAIL.COM>
Date:   Tue Mar 9 21:18:30 2021 -0300

    fix postfixing on compound literal
```

**Comment by andrewvaughanj** [comment]
Likely to be the same issue, but with "more interesting" error messages

#### GCC version:
```bash
gcc (SUSE Linux) 10.2.1 20201202 [revision e563687cf9d3d1278f45aaebd03e0f66531076c9]
```

#### clang version:
```bash
clang version 11.0.1
```

#### Output of `gcc -Werror -Wall -Wextra -Wpedantic -pedantic -std=c11 -c $FNAME`
```bash
(return code 0)
```

#### Output of `clang -Werror -Wall -Wextra -Wpedantic -pedantic -std=c11 -c $FNAME`
```bash
(return code 0)
```

#### Output of `cnip $FNAME`
```bash
v2.c:5:2 warning: missing type specifier, assume `int'
  foo(&reg_stat, 0, __builtin_offsetof(struct reg_stat, sign_bit_copies));
  ^
v2.c:5:6 warning: missing type specifier, assume `int'
  foo(&reg_stat, 0, __builtin_offsetof(struct reg_stat, sign_bit_copies));
      ^
v2.c:5:6 error: expected `)', got `&'
  foo(&reg_stat, 0, __builtin_offsetof(struct reg_stat, sign_bit_copies));
      ^

(return code 0)
```

#### Content of file
```bash
struct reg_stat {
  char sign_bit_copies;
} reg_stat;
void *foo(void *, int, int);
void __attribute__init_reg_last() {
  foo(&reg_stat, 0, __builtin_offsetof(struct reg_stat, sign_bit_copies));
}
```

#### `psychec` version
```bash
commit d10dc1ddfb7ea845f59274a4d6ca5d021ab31423
Author: Leandro T. C. Melo <LTCMELO@GMAIL.COM>
Date:   Tue Mar 9 21:18:30 2021 -0300

    fix postfixing on compound literal
```

**Comment by ltcmelo** [comment]
Labeling as a [C extension](https://gcc.gnu.org/onlinedocs/gcc-11.2.0/gcc/Offsetof.html#Offsetof).

**Comment by ltcmelo** [comment]
Fixed by #82
